src/coord/sink_connector.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::time::Duration;
+
+use failure::{bail, format_err, ResultExt};
+use rdkafka::admin::{AdminClient, AdminOptions, NewTopic, TopicReplication};
+use rdkafka::config::ClientConfig;
+
+use dataflow_types::{
+    KafkaSinkConnector, KafkaSinkConnectorBuilder, SinkConnector, SinkConnectorBuilder,
+};
+use expr::GlobalId;
+use ore::collections::CollectionExt;
+
+pub async fn build(
+    builder: SinkConnectorBuilder,
+    id: GlobalId,
+) -> Result<SinkConnector, failure::Error> {
+    match builder {
+        SinkConnectorBuilder::Kafka(k) => build_kafka(k, id).await,
+    }
+}
+
+async fn build_kafka(
```

Comment:
I know this is a pain but do we want to revert the schema registry insert if topic creation fails?

Reply:
Ugh, good point. I reordered this function to at least create the Kafka topic first, but we might now leak the Kafka topic if we're unable to publish to the schema registry. I think this order is a little better, as there's an expectation that if you can create a Kafka topic with a given name you can publish a schema to the corresponding subject, but not vice-versa.

Left a todo about deleting the Kafka topic. It'd be best effort anyway since we can't transactionally write to both Kafka and the schema registry, so I'm not too worried about leaking the topic.

Reply:
that sounds good to me!